File: Swift/백준/Silver/2217. 로프/로프.swift
if let line = readLine(), let n = Int(line) {
    
    var ropes = [Int]()
    for _ in 0..<n {
        if let input = readLine(), let weight = Int(input) {
            ropes.append(weight)
        }
    }

    ropes.sort(by: >)

    var maxWeight = 0

    for i in 0..<n {
        let currentWeight = ropes[i] * (i + 1)
        
        if currentWeight > maxWeight {
            maxWeight = currentWeight
        }
    }

    print(maxWeight)
}

File: Java/백준/Silver/2217. 로프/로프.java
import java.util.*;

public class Main{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        sc.nextLine();
        
        Integer[] ropes = new Integer[n];
        for (int i = 0; i<n; i++) {
            int input = sc.nextInt();
            sc.nextLine();
            ropes[i] = input;
        }

        Arrays.sort(ropes, Collections.reverseOrder());

        int maxWeight = Integer.MIN_VALUE;
        for (int i = 0; i<n; i++) {
            int currentWeight = ropes[i] * (i+1);

            if (currentWeight > maxWeight) {
                maxWeight = currentWeight;
            }
        }

        System.out.println(maxWeight);
    }
}